netneurotools/metrics.py

```diff
@@ -99,3 +101,87 @@ def communicability_wei(adjacency):
     cmc[np.diag_indices_from(cmc)] = 0
 
     return cmc
+
+
+def rich_feeder_peripheral(x, sc, stat='median'):
+    """
+    Calculates median or mean "connectivity" in rich (hub to hub), feeder
+    (hub to non-hub), and peripheral (non-hub to non-hub) links.
+
+    This function takes a square symmetric correlation/connectivity matrix `x`
+    and computes the median or mean value within rich, feeder, and peripheral
+    links. Hubs are defined againt a threshold (from 0 to the maximum degree
+    `k`, defined on the structural connectivity matrix, `sc`).
+
+    Parameters
+    ----------
+    x : (N, N) numpy.ndarray
+        Symmetric correlation or connectivity matrix
+    sc : (N, N) numpy.ndarray
+        binary structural connectivity matrix
+    stat : str
+        statistic to use over rich/feeder/peripheral links
+        'mean' or 'median' (default)
+
+    Returns
+    -------
+    rfp : (3, k) numpy.ndarray
+        Array of median rich (0), feeder (1), and peripheral (2)
+        values, defined by `x`. `k` is the maximum degree defined on `sc`.
+    pvals : (3, k) numpy.ndarray
+        p-value for each link, computed using Welch's t-test.
+        Rich links are compared against non-rich links. Feeder links are
+        compared against peripheral links. Peripheral links are compared
+        against feeder links. T-test is two-sided.
+
+    Author
+    ------
+    This code was written by Justine Hansen who promises to fix and even
+    optimize the code should any issues arise, provided you let her know.
+    """
+
+    nnodes = len(sc)
+    mask = np.triu(np.ones(nnodes), 1) > 0
+    node_degree = degrees_und(sc)
+    k = np.max(node_degree).astype(np.int64)
+    rfp_label = np.zeros((len(sc[mask]), k))
+
+    for i in range(k):                        # for each degree threshold
+        hub_idx = np.where(node_degree >= i)  # find the hubs
+        hub = np.zeros([nnodes, 1])
+        hub[hub_idx, :] = 1
+
+        rfp = np.zeros([nnodes, nnodes])       # for each link, define rfp
+        for ii in range(nnodes):
+            for iii in range(nnodes):
```

Comment:
This is a bit picky / stylistic, but I (tend to) prefer to have slightly more descriptive loop variables. Could you replace `ii` and `iii` with either `edge1` and `edge2` (or something that provides a bit more clarity about what they are?)